src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,187 @@
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Query parameters are generated for GET methods in Table/Views.
+    /// $select, $filter, $orderby, $first, $after are the query parameters.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    /// <param name="entitySourceType">The source type of the entity.</param>
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table with query parameters")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View with query parameters")]
+    public async Task ValidateQueryParametersForTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+
+        foreach (OpenApiPathItem pathItem in openApiDocument.Paths.Values)
+        {
+            foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
+            {
+                if (operationType is OperationType.Get)
+                {
+                    Assert.IsTrue(operation.Parameters.Count is 5);
+
+                    // Assert that it contains all the query parameters with appropriate type
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$select") && param.Schema.Type.Equals("string")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$filter") && param.Schema.Type.Equals("string")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$orderby") && param.Schema.Type.Equals("string")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$first") && param.Schema.Type.Equals("integer")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$after") && param.Schema.Type.Equals("string")));
+                }
+                else
+                {
+                    // No query parameters for other OperationTypes. 
+                    Assert.IsFalse(operation.Parameters.Any(param => param.In is ParameterLocation.Query));
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Input parameters are generated for Stored Procedures with GET operation.
+    /// Input parameters for stored procedures will be generated for GET operation.
+    /// If a parameter default value is not provided in the config, it will marked as required.
```

Comment:
```suggestion
    /// If a parameter default value is not provided in the config, it will be marked as required.
```
nit

Reply:
this may also change with Rohan's latest PR affecting input parameters validation and expected presence for stored procedures.